x/evm/keeper/querier.go

```diff
@@ -273,17 +274,16 @@ func QueryDepositAddress(ctx sdk.Context, k types.ChainKeeper, n types.Nexus, da
 		return nil, sdkerrors.Wrap(types.ErrEVM, fmt.Sprintf("token for asset '%s' not confirmed", params.Asset))
 	}
 
-	depositAddr, _, err := k.GetBurnerAddressAndSalt(ctx, token.GetAddress(), params.Address, gatewayAddr)
-	if err != nil {
-		return nil, err
-	}
-
-	_, ok = n.GetRecipient(ctx, nexus.CrossChainAddress{Chain: depositChain, Address: depositAddr.String()})
+	recipientChain, ok := n.GetChain(ctx, params.Chain)
 	if !ok {
-		return nil, sdkerrors.Wrap(types.ErrEVM, "deposit address is not linked with recipient address")
+		return nil, sdkerrors.Wrap(types.ErrEVM, fmt.Sprintf("%s is not a registered chain", params.Chain))
 	}
 
-	return depositAddr.Bytes(), nil
+	recipient := nexus.CrossChainAddress{Chain: recipientChain, Address: params.Address}
+	addresses := k.GetBurnerAddresses(ctx, recipient)
+	resp := types.QueryAddressesResponse{Addresses: addresses}
+
+	return types.ModuleCdc.MarshalLengthPrefixed(&resp)
```

Comment:
Don't you also need to change the code in https://github.com/axelarnetwork/axelar-core/blob/7d7eca2851e85bfeee5d7a137c11960482d83c47/x/evm/client/cli/query.go?

Reply:
you are right, great call!

Reply:
done